lib/serviceClient.ts

```diff
@@ -240,12 +240,24 @@ export class ServiceClient {
     let httpPipeline: RequestPolicy = this._httpClient;
     if (this._requestPolicyFactories && this._requestPolicyFactories.length > 0) {
       for (let i = this._requestPolicyFactories.length - 1; i >= 0; --i) {
-        httpPipeline = this._requestPolicyFactories[i].create(
+        const nextPolicyPipeline = this._requestPolicyFactories[i].create(
           httpPipeline,
           this._requestPolicyOptions
         );
+
+        if (httpRequest.redirectLimit !== undefined && nextPolicyPipeline instanceof RedirectPolicy) {
+          // if redirect behaviour is specified on the request, we don't want the client's default redirectPolicy to apply
+          continue;
+        }
+
+        httpPipeline = nextPolicyPipeline;
       }
     }
+
+    if (httpRequest.redirectLimit !== undefined) {
+      httpPipeline = new RedirectPolicy(httpPipeline, this._requestPolicyOptions, httpRequest.redirectLimit);
```

Comment:
I don't like this implementation. It assumes every request **must** use RedirectPolicy, and it questionably changes the ordering so redirect policy is first.

Reply:
The redirectPolicy is **only** applied if a redirectLimit is specificed. If the redirectLimit is specified, then it must be applied as it is the redirectPolicy that counts and limits the redirects.
The ordering is an interesting question. In my testing, placing it first works as expected. If it's exact position in the pipeline becomes important for some othe reason, this code could be extended to support that.

Reply:
Why not just make `redirectPolicy` smart enough to skip itself if `redirectLimit` isn't defined? Why make `ServiceClient` have the burden of knowing the interactions between specific policies?

Reply:
That is so obvious now that you've said it. I've removed all of the changes to ServiceClient.